packages/core/src/signer/types.ts

```diff
@@ -1,12 +1,23 @@
 import type { Address } from "abitype";
-import type { Hash, Hex } from "viem";
+import type { Hex } from "viem";
 import type { SignTypedDataParams } from "../account/types.js";
 
-export interface SmartAccountSigner {
+export interface AuthSmartAccountSigner<Inner, AuthParams, AuthDetails>
+  extends SmartAccountSigner<Inner> {
+  authenticate: (params: AuthParams) => Promise<AuthDetails>;
+
+  getAuthDetails: () => Promise<AuthDetails>;
+}
+
+export interface SmartAccountSigner<Inner> {
```

Comment:
if we default this to `Inner = any` does that simplify a lot of the breaking changes?